tests/mir-opt/issue_99325.main.built.after.mir

```diff
@@ -1,8 +1,8 @@
 // MIR for `main` after built
 
 | User Type Annotations
-| 0: user_ty: Canonical { value: TypeOf(DefId(0:3 ~ issue_99325[22bb]::function_with_bytes), UserArgs { args: [Const { ty: &'static [u8; 4], kind: Branch([Leaf(0x41), Leaf(0x41), Leaf(0x41), Leaf(0x41)]) }], user_self_ty: None }), max_universe: U0, variables: [] }, span: $DIR/issue_99325.rs:10:16: 10:46, inferred_ty: fn() -> &'static [u8] {function_with_bytes::<&*b"AAAA">}
-| 1: user_ty: Canonical { value: TypeOf(DefId(0:3 ~ issue_99325[22bb]::function_with_bytes), UserArgs { args: [Const { ty: &'static [u8; 4], kind: UnevaluatedConst { def: DefId(0:8 ~ issue_99325[22bb]::main::{constant#1}), args: [] } }], user_self_ty: None }), max_universe: U0, variables: [] }, span: $DIR/issue_99325.rs:11:16: 11:68, inferred_ty: fn() -> &'static [u8] {function_with_bytes::<&*b"AAAA">}
+| 0: user_ty: Canonical { value: TypeOf(DefId(0:3 ~ issue_99325[22bb]::function_with_bytes), UserArgs { args: [Const { ty: &ReStatic [u8; Const { ty: usize, kind: Leaf(0x0000000000000004) }], kind: Branch([Leaf(0x41), Leaf(0x41), Leaf(0x41), Leaf(0x41)]) }], user_self_ty: None }), max_universe: U0, variables: [] }, span: $DIR/issue_99325.rs:10:16: 10:46, inferred_ty: fn() -> &'static [u8] {function_with_bytes::<&*b"AAAA">}
```

Comment:
The previous display rendering is bit size independent.

Also having def ids in tests is going to cause a lot of churn on those tests. While you could filter away these changes, I'd prefer if they were not using debug rendering.

Reply:
Considering the other debug rendering around the type here, maybe temporarily using filters to get this PR landed is best. A follow up could then change `user_ty` to use some better pretty printing

Reply:
About the def ids: the old output also prints them, right? I haven't changed that.

Reply:
Why does the `Leaf` value change depending on the platform word size? It's a `ScalarInt`, which contains a `u128`, which seems like it should be printed in a platform-independent fashion...

Reply:
To answer my own question: it's because of [this](https://github.com/rust-lang/rust/blob/c0583a0221c28f01a7bdbc01464baf22f7c6d4a8/compiler/rustc_middle/src/ty/consts/int.rs#L482-L489).

Reply:
I have used `EMIT_MIR_FOR_EACH_BIT_WIDTH` to handle the differing output on 32-bit and 64-bit. That seems to be the standard way to handle such differences.